addons/l10n_sa_edi/models/account_journal.py

```diff
@@ -626,16 +626,20 @@ def _l10n_sa_call_api(self, request_data, request_url, method):
         response_data['status_code'] = request_response.status_code
 
         val_res = response_data.get('validationResults', {})
-        if not request_response.ok and (val_res.get('errorMessages') or val_res.get('warningMessages')):
+        if not request_response.ok and (val_res.get('errorMessages') or val_res.get('warningMessages') or response_data.get('errors')):
             error = "" if not status_code else Markup("<b>[%s]</b>") % (status_code)
-            if isinstance(response_data, dict) and val_res.get('errorMessages'):
-                error += _("Invoice submission to ZATCA returned errors")
-                return {
-                    'error': error,
-                    'json_errors': response_data,
-                    'blocking_level': 'error',
-                }
-            error += request_response.reason
+            if not isinstance(response_data, dict):
+                if val_res.get('errorMessages'):
+                    return {
+                        'error': error + _("Invoice submission to ZATCA returned errors"),
+                        'json_errors': response_data,
+                        'blocking_level': 'error',
+                    }
+                elif response_data.get('errors'):
+                    return {
+                        'error': error + response_data["errors"][0].get('message', _("Unknown error")),
+                        'json_errors': response_data,
+                    }
             return {
                 'error': error,
                 'blocking_level': 'error',
```

Comment:
Maybe you can just add response_data here in an extra key and treat it after when this method is called for the OTP?  The two work a bit different, but called through the same method...